src/main/java/io/supertokens/useridmapping/UserIdMapping.java

```diff
@@ -70,7 +123,53 @@ public static io.supertokens.pluginInterface.useridmapping.UserIdMapping getUser
 
     public static boolean deleteUserIdMapping(Main main, String userId, UserIdType userIdType)
             throws StorageQueryException {
+        return deleteUserIdMapping(main, userId, userIdType, false);
+    }
+
+    public static boolean deleteUserIdMapping(Main main, String userId, UserIdType userIdType, boolean force)
+            throws StorageQueryException {
+        // if a userIdMapping is deleted with force, then we skip the following checks
+        if (!force) {
+            String externalId;
+            if (userIdType == UserIdType.EXTERNAL) {
+                externalId = userId;
+            } else {
+                io.supertokens.pluginInterface.useridmapping.UserIdMapping userIdMapping = getUserIdMapping(main,
+                        userId, UserIdType.ANY);
+                if (userIdMapping != null) {
+                    externalId = userIdMapping.externalUserId;
+                } else {
+                    return false;
+                }
+            }
+            // check if externalId is used in any non-auth recipes
+            {
+                if (StorageLayer.getStorage(main).isUserIdBeingUsedInNonAuthRecipe(SessionStorage.class.getName(),
+                        externalId)) {
+                    throw new IllegalStateException("External Id is already in use in Session recipe");
+                }
+            }
+
+            {
+                if (StorageLayer.getStorage(main).isUserIdBeingUsedInNonAuthRecipe(UserMetadataStorage.class.getName(),
```

Comment:
Extract this whole thing into a common function which this function and the create mapping function can check